### DISCUSSION TITLE
What is the correct way of using a static / assets folder with Parcel 2?

### DISCUSSION THREAD

**Discussion by halvardssm** [parcel-bundler/parcel#4716] [General] [answered]
As the title asks. I have a large number of static files (images, logos etc.) which I would like to include and make available by referencing them directly in the react-dom. Example:
```tsx
render(){
  return <div><img src="static/logo.png" /></div>
}
```
Another use-case is when I want to reference the assets programmatically:
```tsx
export const FunctionalImageComponent = ({ src: string }) => <img src={src} />
...
render(){
  return <FunctionalImageComponent src="static/logo.png" />
}
```

Thanks in advance!

**Comment by mischnic** [comment, marked as answer]
Parcel 1: 
```js
import logo from "static/logo.png";

render(){
  return <div><img src={logo} /></div>
}
```

Parcel 2: 
```js
import logo from "url:static/logo.png";

render(){
  return <div><img src={logo} /></div>
}
```

(The full path of the asset has to be statically analyzable)

**Reply by halvardssm** [reply]
Thanks! Will this also work with dynamic import and wildcard import?
```tsx
render(){
  const logo = import(someVariable);
  return <div><img src={logo} /></div>
}
```

```tsx
render(){
  const logos = import(`url:${folderDir}/*.png`);
  return <div>
    {logos.map(el=><img src={logo} />)}
  </div>
}
```

**Reply by mischnic** [reply]
No, it has to be a ("constant") string literal

**Reply by halvardssm** [reply]
Then it is unfortunately what I am looking for... I need to dynamically fetch files in the project I am working on and exposing a static assets folder would be the best way to do it. Thanks anyway!

**Reply by mischnic** [reply]
(Parcel 1 supports globs `import files from "./images/*.png";`, which might work if you have only a few possible `folderDir` values)

**Reply by halvardssm** [reply]
Right, but that will require me to import the whole folder during the initial render, and then pass along the files through my react component props... You only mentioned Parcel 1, does that mean it's not planned/working for Parcel 2? Is there no other way of linking a static folder? I have also tried to copy the static folder into the dist before `parcel serve` but this doesn't seem like the right approach either 🤔

**Reply by mischnic** [reply]
Follow https://github.com/parcel-bundler/parcel/issues/4683 regarding globs in Parcel 2

>  I have also tried to copy the static folder into the dist before parcel serve but this doesn't seem like the right approach either 🤔

That should work

**Reply by ghost** [reply]
I am getting this when importing a .png with `url:...`:

```bash
Cannot find module 'url:../static/parcel.png' or its corresponding type declarations.ts(2307)
```

> (The full path of the asset has to be **statically analyzable**)

What do you mean by statically analyzable?
